### PR TITLE
convert string to fs::path

### PR DESCRIPTION
**Pull request by neheb**
Avoids conversions

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/url_request_handler.cc [hunk 1]
```diff
@@ -70,11 +70,11 @@ void URLRequestHandler::getInfo(const char* filename, UpnpFileInfo* info)
         UpnpFileInfo_set_FileLength(info, -1);
     } else {
 #ifdef ONLINE_SERVICES
-        std::string url = item->getFlag(OBJECT_FLAG_ONLINE_SERVICE) ? OnlineServiceHelper::resolveURL(item) : item->getLocation().string();
+        fs::path url = item->getFlag(OBJECT_FLAG_ONLINE_SERVICE) ? OnlineServiceHelper::resolveURL(item) : item->getLocation();
```
Comment (KarlStraussberger) [inline src/url_request_handler.cc:73]: It's a URL so it starts with `http(s)://`. We should not declare that as path.

File: src/content/onlineservice/online_service_helper.cc [hunk 2]
```diff
@@ -36,7 +36,7 @@
 #include "config/config_manager.h"
 #include "online_service.h"
 
-std::string OnlineServiceHelper::resolveURL(const std::shared_ptr<CdsItemExternalURL>& item)
+fs::path OnlineServiceHelper::resolveURL(const std::shared_ptr<CdsItemExternalURL>& item)
```
Comment (KarlStraussberger) [inline src/content/onlineservice/online_service_helper.cc:39]: It's a URL so it starts with `http(s)://`. We should not declare that as path.

File: src/url_request_handler.cc (hunk 1 above)
Comment (neheb) [reply src/url_request_handler.cc:73]: What's interesting is getLocation() returns an fs::path

Comment (KarlStraussberger) [reply src/url_request_handler.cc:73]: It's the special situation of external items which inherit from CdsObject we have to live with, but we should make it an URL here (is there some additional lib for that?).

**Pull request merged**